applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -89,12 +101,83 @@ void GeoLinearElasticPlaneStrain2DLaw::CalculatePK2Stress(const Vector& rStrainV
 {
     KRATOS_TRY
 
+    UpdateInternalDeltaStrainVector(rValues.GetStrainVector());
+
     Matrix C;
     this->CalculateElasticMatrix(C, rValues);
-    // Total formulation
-    noalias(rStressVector) = prod(C, rStrainVector);
+
+    // Incremental formulation
+    noalias(mStressVector) = mStressVectorFinalized + prod(C, mDeltaStrainVector);
+
+    SetExternalStressVector(rStressVector);
 
     KRATOS_CATCH("")
 }
 
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalDeltaStrainVector(const Vector& r_strain_vector){
+    mDeltaStrainVector = r_strain_vector - mStrainVectorFinalized;
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalStrainVectorFinalized(const Vector& r_strain_vector) {
+    this->SetInternalStrainVector(r_strain_vector);
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetExternalStressVector(Vector& r_stress_vector) const{
+    r_stress_vector = mStressVector;
```

Comment:
I'm not sure assignment like this will work like intended. I think in this way, the r_stress_vector will actually become the (non-const) ref to the member, is that the intention, or do we want to set the values of r_stress_vector using the ones of mStressvector? Same holds for the other assignment functions.

Edit: Starting to doubt, since the mStressVector is not a reference itself, so it might correctly copy mStressvector data to the r_stress_vector

@avdg81 Would you know the answer here?